### ISSUE TITLE
Ignore Hugo Markdown attributes

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#756]
### Problem or motivation

For Markdown that's intended to be rendered by Hugo, some will contain "Markdown attributes" (https://gohugo.io/content-management/markdown-attributes/).

These must be hard up against the content they affect, which when they're treated like normal content, can cause them to be flagged.

For example, MD058 (https://github.com/rvben/rumdl/blob/e06f03d515fc81054e3929b25d252020960a9c11/docs/md058.md) requires tables have blank lines around them. However, if the Markdown table has a "Markdown attribute" against it, it will be flagged.

### Proposed solution

Ideally, the Markdown provided in this feature request wouldn't affect MD058. Markdown attributes would, ideally, not be considered content for the purposes of any rule requires blank lines around things.

### Feature category

Other

### Example markdown

```markdown
---
title: Level 1 heading
---

This is a table:

| Column 1 | Column 2 |
|:---------|:---------|
| Row 1    | Row 1    |
| Row 2    | Row 2    |
{class="a" id="b"}
```

**Comment by rvben** [comment]
Added in v0.2.41.

rumdl now has a `hugo` flavor (alias `goldmark`) that recognizes Goldmark block attribute lists. Under it, a standalone attribute line placed hard against a block is treated as belonging to that block rather than as content, so it no longer trips the blank-line rules. This covers:

- MD058 (tables) - your reported case
- MD022 (headings)
- MD031 (fenced code blocks)

Enable it globally:

```toml
[global]
flavor = "hugo"   # or "goldmark"
```

or per file:

```toml
[per-file-flavor]
"content/**/*.md" = "hugo"
```

Your example is clean under the Hugo flavor, while it still flags under `standard`, where `{class="a"}` is literal text that legitimately needs a surrounding blank line:

```
$ rumdl check --flavor hugo test.md
Success: No issues found in 1 file

$ rumdl check --flavor standard test.md
test.md:10:24: [MD058] Missing blank line after table
```

Both the bare Goldmark form (`{class="a" id="b"}`) and the kramdown IAL form (`{:.class}`) are recognized.

One deliberate exclusion: lists (MD032). A bare attribute line directly after a list item is a CommonMark lazy paragraph continuation, so it is absorbed into the list and never flagged in any flavor. There was nothing to fix there. If you hit a real case where an attribute list on a list trips a rule, let me know.

Docs: https://github.com/rvben/rumdl/blob/main/docs/flavors/hugo.md

Thanks for the report and the example.

**Comment by eread** [comment]
Thanks @rvben. That's a comprehensive implementation! I guess we'll want an update to: https://github.com/SchemaStore/schemastore/blob/1817138a314f4e3d0b1d40a0327e6e3753c976dc/src/schemas/json/rumdl.json#L179 also?

Thanks again!

**Comment by eread** [comment]
> Thanks [@rvben](https://github.com/rvben). That's a comprehensive implementation! I guess we'll want an update to: https://github.com/SchemaStore/schemastore/blob/1817138a314f4e3d0b1d40a0327e6e3753c976dc/src/schemas/json/rumdl.json#L179 also?
> 
> Thanks again!

@rvben One other thing for consideration. Perhaps support for HTML comments to sit between content and Markdown attribute? For example:

```markdown
---
title: Level 1 heading
---

This is a table:

| Column 1 | Column 2 |
|:---------|:---------|
| Row 1    | Row 1    |
| Row 2    | Row 2    |
<!-- Explanation of Markdown attribute -->
{class="a" id="b"}
```

At the moment, the HTML comment breaks the support for the Markdown attribute.

**Comment by rvben** [comment]
Thanks @eread, good questions on both.

**SchemaStore:** that update is automated. rumdl's release pipeline regenerates the SchemaStore-format schema and opens a PR against SchemaStore after each release, so the new `hugo`/`goldmark` values are already on their way in SchemaStore/schemastore#6109 (open, checks green, just waiting on a SchemaStore maintainer to merge). Once it lands, editors validating against SchemaStore will accept `flavor = "hugo"`. Nothing extra needed on the rumdl side.

**HTML comment between the table and the attribute list:** I dug into this, and it turns out Hugo itself doesn't attach the attribute when a comment sits in between. With `[markup.goldmark.parser.attribute] block = true`, Hugo attaches `{...}` to the block immediately above it. When that block is an HTML comment, the attribute attaches to the comment (which can't render it) and is silently dropped, so the table gets no class/id:

```markdown
| Column 1 | Column 2 |
|:---------|:---------|
| Row 1    | Row 1    |
<!-- Explanation -->
{class="a" id="b"}
```

renders as a plain `<table>` (no attributes) in Hugo v0.164.0.

So MD058 firing there is actually pointing at a real problem: that layout won't do what you intend in Hugo, because the attribute list has to sit hard against the block.

The good news is you can keep the explanatory comment by putting it right after the attribute line instead:

```markdown
| Column 1 | Column 2 |
|:---------|:---------|
| Row 1    | Row 1    |
{class="a" id="b"}
<!-- Explanation of Markdown attribute -->
```

That renders as `<table class="a" id="b">` in Hugo, and rumdl (with `flavor = "hugo"`) is clean on it too.

Given Hugo's own behavior, I don't think rumdl should suppress the warning for the comment-in-between case, since that would mask markup that silently loses the attribute. If you run into a case where Hugo does attach an attribute across a comment, let me know and I'll take another look.

**Comment by eread** [comment]
> So MD058 firing there is actually pointing at a real problem: that layout won't do what you intend in Hugo, because the attribute list has to sit hard against the block.

@rvben I've dug back into that you're correct! It turns out, MD058 does point to a real problem which is great. Leaving it how it is is preferred!

Thanks again.
